test/javascripts/adapaters/topic-list-test.js.es6

```diff
@@ -0,0 +1,11 @@
+module("adapter:topic-list");
+
+import { finderFor } from 'discourse/adapters/topic-list';
+
+test("finderFor", function() {
+  var mock = sandbox.mock(Discourse);
+  mock.expects("ajax").withArgs("/search.json?q=test%25%25");
```

Comment:
Wow thanks for going above and beyond on this one. I think in this case the mock is fair, but could you add a comment explaining why it has to be a mock? Just so someone doesn't come in and try to refactor it and go through the same pain?